packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -22,36 +21,46 @@ import { E } from '@endo/far';
 import { M } from '@endo/patterns';
 import { TxStatus, TxType } from './constants.js';
 import type {
-  CCTPSettlementArgs,
-  CCTPSettlementOfferArgs,
-  CCTPTransactionKey,
   PublishedTx,
+  TransactionSettlementOfferArgs,
+  TxId,
+} from './types.js';
+import {
+  ResolverOfferArgsShapes,
+  TransactionSettlementOfferArgsShape,
 } from './types.js';
-import { CCTPSettlementArgsShape, ResolverOfferArgsShapes } from './types.js';
 
-type CCTPTransactionEntry = {
+type TransactionEntry = {
   destinationAddress: AccountId;
   amountValue: bigint;
   vowKit: VowKit<void>;
+  type: TxType;
 };
 
 const trace = makeTracer('Resolver');
 
-const ClientFacetI = M.interface('CCTPResolverClient', {
-  registerCCTPTransaction: M.call(M.string(), M.nat()).returns(VowShape),
+const ClientFacetI = M.interface('ResolverClient', {
+  registerTransaction: M.call(
+    M.or(...Object.values(TxType)),
+    M.string(),
+    M.nat(),
+  ).returns(VowShape),
 });
 
 const ReporterI = M.interface('Reporter', {
-  insertPendingTransaction: M.call(M.bigint(), M.string()).returns(),
-  completePendingTransaction: M.call(M.string(), M.string()).returns(),
+  insertPendingTransaction: M.call(M.string(), M.string(), M.nat()).returns(),
+  completePendingTransaction: M.call(
+    M.string(),
+    M.or(TxStatus.SUCCESS, TxStatus.FAILED),
```

Comment:
I've seen discussion of a "timed out" status. Consider:

```suggestion
    M.or(...Object.values(TxStatus)),
```

Reply:
oh... I see... we need to leave out "pending".